src/cloudsc_cuda/CMakeLists.txt

```
@@ -191,6 +192,64 @@ if( HAVE_CLOUDSC_C_CUDA )
     )
     ###
 
+    ###### SCC-CUDA-OPT ####
+    ecbuild_add_library(
+        TARGET dwarf-cloudsc-c-cuda-opt-lib
+        INSTALL_HEADERS LISTED
+        SOURCES
+            cloudsc/yoecldp_c.h
+            cloudsc/load_state.h
+            cloudsc/load_state.cu
+            cloudsc/cloudsc_c_opt.h
+            cloudsc/cloudsc_c_opt.cu
+            cloudsc/cloudsc_driver_opt.h
+            cloudsc/cloudsc_driver_opt.cu
+            cloudsc/cloudsc_validate.h
+            cloudsc/cloudsc_validate.cu
+            cloudsc/mycpu.h
+            cloudsc/mycpu.cu
+        PUBLIC_INCLUDES
+            $<INSTALL_INTERFACE:include>
+            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/cloudsc>
+        PUBLIC_LIBS
+            $<${HAVE_HDF5}:hdf5::hdf5>
+            $<${HAVE_SERIALBOX}:Serialbox::Serialbox_C>
+            $<${HAVE_OMP}:OpenMP::OpenMP_C>
+        DEFINITIONS
+            ${CLOUDSC_DEFINITIONS}
+    )
+
+    target_include_directories(
+      dwarf-cloudsc-c-cuda-opt-lib
+      PUBLIC
+      ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}
+    )
+    if (NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
+      target_compile_options(dwarf-cloudsc-c-cuda-opt-lib PRIVATE $<$<COMPILE_LANGUAGE:CUDA>>)
+    else()
+      target_compile_options(dwarf-cloudsc-c-cuda-opt-lib PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:
+          -O3 -use_fast_math -lineinfo -maxrregcount=128 -gencode arch=compute_${CMAKE_CUDA_ARCHITECTURES},code=sm_${CMAKE_CUDA_ARCHITECTURES}>)
+          # -O0 -g -G -maxrregcount=128 -gencode arch=compute_${CMAKE_CUDA_ARCHITECTURES},code=sm_${CMAKE_CUDA_ARCHITECTURES}>)
```

Maybe switch according to build-type?
```suggestion
      if(CMAKE_BUILD_TYPE STREQUAL "Debug")
        target_compile_options(dwarf-cloudsc-c-cuda-opt-lib PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:-O0 -g -G>)
      else()
        target_compile_options(dwarf-cloudsc-c-cuda-opt-lib PRIVATE 
        $<$<COMPILE_LANGUAGE:CUDA>:-O3 -use_fast_math -lineinfo>)
        endif()
        target_compile_options(dwarf-cloudsc-c-cuda-opt-lib PRIVATE $<$<COMPILE_LANGUAGE:CUDA>:-maxrregcount=128 -gencode arch=compute_${CMAKE_CUDA_ARCHITECTURES},code=sm_${CMAKE_CUDA_ARCHITECTURES}>)
```